packages/protocol/test/common/onboarding.ts

```diff
@@ -0,0 +1,424 @@
+        let totalCost = 0
+
+        const deploymentTx = await mtwDeployer.deploy(
+          user,
+          mtw.address,
+          // @ts-ignore
+          mtw.contract.methods.initialize(ensureLeading0x(user)).encodeABI()
+        )
+        logCost(`Deploying and initializing a proxy takes ${deploymentTx.receipt.gasUsed} gas`)
+        totalCost += deploymentTx.receipt.gasUsed
+        mtw = await MTW.at(deploymentTx.logs[0].args.wallet)
+
+        let { v, r, s } = await getParsedSignatureOfAddress(web3, mtw.address, user)
+        const setAccount = {
+          value: 0,
+          destination: accounts.address,
+          // @ts-ignore
+          data: accounts.contract.methods.setAccount('', dek, user, v, r, s).encodeABI(),
+          nonce: 0,
+        }
+        const setAccountTx = await executeMetaTransaction(user, mtw, setAccount)
+        logCost(`Setting the account takes ${setAccountTx.receipt.gasUsed} gas`)
+        totalCost += setAccountTx.receipt.gasUsed
+
+        await stableToken.transfer(relayermtw.address, attestationFee)
+
+        const approve = {
+          value: 0,
+          destination: stableToken.address,
+          // @ts-ignore
+          data: stableToken.contract.methods
+            .approve(attestations.address, attestationFee)
+            .encodeABI(),
+          nonce: 1,
+        }
+        const approveMeta = {
+          value: 0,
+          destination: mtw.address,
+          data: await getExecuteMetaTransactionData(user, mtw, approve),
+        }
+
+        const request = {
+          value: 0,
+          destination: attestations.address,
+          // @ts-ignore
+          data: attestations.contract.methods
+            .request(identifier, numAttestations, stableToken.address)
+            .encodeABI(),
+          nonce: 2,
+        }
+        const requestMeta = {
+          value: 0,
+          destination: mtw.address,
+          data: await getExecuteMetaTransactionData(user, mtw, request),
+        }
+
+        const transfer = {
+          value: 0,
+          destination: stableToken.address,
+          // @ts-ignore
+          data: stableToken.contract.methods.transfer(mtw.address, attestationFee).encodeABI(),
+        }
+
+        transactions = [transfer, approveMeta, requestMeta]
+        const requestTx = await relayermtw.executeTransactions(
+          transactions.map((t) => t.destination),
+          transactions.map((t) => t.value),
+          ensureLeading0x(transactions.map((t) => trimLeading0x(t.data)).join('')),
+          transactions.map((t) => trimLeading0x(t.data).length / 2)
+        )
+        logCost(`Requesting attestations takes ${requestTx.receipt.gasUsed} gas`)
+        totalCost += requestTx.receipt.gasUsed
+
+        // Add fake randomness so that issuers can be selected.
+        const requestBlockNumber = await web3.eth.getBlockNumber()
+        await random.addTestRandomness(requestBlockNumber + selectIssuersWaitBlocks, '0x1')
```

Comment:
using this randomness in `selectIssuers` uses as much gas as reading real randomness
can you confirm?

Reply:
I cannot confirm and suspect that's not exactly the case but I don't think it matters